plugins/woocommerce/client/blocks/assets/js/atomic/blocks/product-elements/price/frontend.ts

```diff
@@ -0,0 +1,49 @@
+/**
+ * External dependencies
+ */
+import { getContext, getElement, store } from '@wordpress/interactivity';
+import type { DisplayedProduct } from '@woocommerce/type-defs/product';
+import type { SingleProductTemplateStore } from '@woocommerce/base-stores/single-product-template';
+
+/**
+ * Internal dependencies
+ */
+
+// Stores are locked to prevent 3PD usage until the API is stable.
+const universalLock =
+	'I acknowledge that using a private store means my plugin will inevitably break on the next store release.';
+
+const { state: wooState } = store< SingleProductTemplateStore >(
+	'woocommerce',
+	{},
+	{ lock: universalLock }
+);
+
+const productPriceStore = store(
+	'woocommerce/product-price',
+	{
+		callbacks: {
+			setNewPrice: () => {
+				const element = getElement();
+
+				if ( ! element.ref ) {
+					return;
+				}
+
+				const context = getContext< {
+					displayedProduct: DisplayedProduct;
+				} >( 'woocommerce/single-product' );
+
+				if ( context ) {
+					element.ref.innerHTML = context.displayedProduct.price_html;
+				} else if ( wooState.displayedProduct ) {
+					element.ref.innerHTML =
+						wooState.displayedProduct.price_html;
```

Comment:
We need to sanitize the HTML before setting it. I can take care of it if you agree with the approach. Unfortunately, we have to use the “price HTML” directly, as this part of WooCommerce contains a lot of logic and is highly extensible.

Reply:
Otherwise, we should re-think the Product Price block and move it to a full block-paradigm.

Reply:
Not really, but I'm fine with using `sanitizeHTML` util. At the same time, I would check the impact on the bundle size. According [to bundlephobia](https://bundlephobia.com/package/dompurify@3.2.6), dompurify is pretty heavy, but it is tree-shakeable. I don't know why, but we are importing the entire module:

https://github.com/woocommerce/woocommerce/blob/181e81e3e2c6d5be8b9dd9f0d3034ac50fcced5c/plugins/woocommerce/client/blocks/assets/js/utils/sanitize-html.ts#L9-L21

We could import only `sanitize` to reduce the size. 


Also, I think this is a good interim solution, but we should avoid to set HTML even from the extensibility point of view.

Reply:
Done in https://github.com/woocommerce/woocommerce/pull/58384/commits/a43523724dc4f2abb10303d84e25e7d1f0b46bb8.